be/src/storage/lake/vacuum.cpp

```diff
@@ -72,6 +73,33 @@ std::future<Status> completed_future(Status value) {
     return p.get_future();
 }
 
+bool should_retry(const Status& st, int64_t attempted_retries) {
+    if (attempted_retries >= config::lake_vacuum_retry_max_attempts) {
+        return false;
+    }
+    Slice message = st.message();
+    return MatchPattern(StringPiece(message.data, message.size), config::lake_vacuum_retry_pattern);
+}
+
+int64_t calculate_retry_delay(int64_t attempted_retries) {
+    int64_t min_delay = config::lake_vacuum_retry_min_delay;
+    return min_delay * (1 << attempted_retries);
+}
+
+Status delete_files_with_retry(FileSystem* fs, const std::vector<std::string>& paths) {
+    for (int64_t attempted_retries = 0; /**/; attempted_retries++) {
+        auto st = fs->delete_files(paths);
```

Comment:
a little afraid of these batch deletes, some of them are deleted and some of them are not, will this batch retry causes qps amplified for those already deleted files?

Reply:
Maybe, but the starlet does not tell which files have been deleted, there is nothing we can do here.

Reply:
I'll submit another patch to limit the batch size.